src/main/java/org/traccar/protocol/Gl200TextProtocolDecoder.java

```diff
@@ -1335,8 +1338,9 @@ private Object decodeDtt(Channel channel, SocketAddress remoteAddress, String se
 
     private Object decodeBaa(Channel channel, SocketAddress remoteAddress, String sentence) {
         Parser parser = new Parser(PATTERN_BAA, sentence);
-        Position position = initPosition(parser, channel, remoteAddress);
-        if (position == null) {
+        Position position = new Position();
+        String model = initPosition(parser, channel, remoteAddress, position);
+        if (model == null) {
```

Comment:
Position should always be created with the protocol argument, except for test cases.

And `model == null` check doesn't really make sense to me.